src/main/java/org/sonarlint/intellij/analysis/AnalysisTask.java

```diff
@@ -284,6 +285,9 @@ private void processRawIssue(LiveIssueBuilder issueBuilder, IssueManager manager
       LiveIssue locallyTrackedIssue = manager.trackSingleIssue(vFile, previousIssues, liveIssue);
       issuesPerFile.computeIfAbsent(vFile, f -> new ArrayList<>()).add(locallyTrackedIssue);
     }
+
+    SonarLintTelemetry telemetry = SonarLintUtils.getService(SonarLintTelemetry.class);
+    telemetry.addReportedRule(liveIssue.getRuleKey());
```

Comment:
I wonder if there is not a risk of performance decrease with such call. On telemetry side, we take a lock on the telemetry file every time we write to it.
I think we should change the API to allow a single call per analysis, where we report all rules in one call.